### PR TITLE
modify cached_gc_safe_point to atomic to prevent more request to PD

### LINKED ISSUE #4928: multi-thread safe getGCSafePointWithRetry
## Enhancement

Currently there is a race condition in `getGCSafePointWithRetry` which makes it buggy. The `getGCSafePointWithRetry` are now called in a least 3 kind of threads:

- Read thread
- Background thread
- Manual compaction thread

### PR DESCRIPTION
**Pull request by shuke987**
### What problem does this PR solve?
safe_point_last_update_time is used to prevent every gc_safe_point request calling PD. But it is not atmoic, thus when getGCSafePointWithRetry calling by multiple threads, it may not worked as expected. Fix this.

Issue Number: ref #4928 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by JaySon-Huang** [comment]
https://ci.pingcap.net/blue/organizations/jenkins/tiflash-build-common/detail/tiflash-build-common/6122/pipeline

need to format the codes by using the `format-diff.py` script https://github.com/pingcap/tiflash#contributing

**Comment by JaySon-Huang** [comment]
I changed it to be `ref #4928` cause we can avoid useless calls to PD https://github.com/pingcap/tiflash/issues/4928#issuecomment-1147275463

File: dbms/src/Storages/Transaction/PDTiKVClient.h [hunk 1]
```diff
@@ -49,7 +51,7 @@ struct PDClientHelper
         {
             // In case we cost too much to update safe point from PD.
             std::chrono::time_point<std::chrono::system_clock> now = std::chrono::system_clock::now();
-            const auto duration = std::chrono::duration_cast<std::chrono::seconds>(now - safe_point_last_update_time);
+            const auto duration = std::chrono::duration_cast<std::chrono::seconds>(now - safe_point_last_update_time.load());
```
Comment (breezewish) [inline dbms/src/Storages/Transaction/PDTiKVClient.h:54]: Maybe using `*` is enough:

```suggestion
            const auto duration = std::chrono::duration_cast<std::chrono::seconds>(now - *safe_point_last_update_time);
```

**Review by breezewish — APPROVED** [review]
The rest LGTM

**Review by Lloyd-Pottiger — APPROVED** [review]
LGTM

File: dbms/src/Storages/Transaction/PDTiKVClient.h (hunk 1 above)
Comment (JaySon-Huang) [reply dbms/src/Storages/Transaction/PDTiKVClient.h:54]: @breezewish there is no `operator *` for atomic type in C++ 🤔 https://en.cppreference.com/w/cpp/atomic/atomic

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**